pennylane/circuit_drawer/drawable_layers.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains a helper function to sort operations into layers.
+"""
+
+from .utils import default_wire_map
+
+
+def _recursive_find_layer(checking_layer, op_occupied_wires, occupied_wires_per_layer):
+    """Determine correct layer for operation with ``op_occupied_wires``
+
+    Args:
+        checking_layer (int): function determines if operation fits on this layer
+        op_occupied_wires (set(int)): wires covered the drawn operation.  Includes everything
+            between used wires in a multi-wire gate.
```

Comment:
Slightly confused here. Would this mean that for `qml.CNOT(wires=[0,2])`:
* `0`;
*  `2`;
* and also `1` are in `op_occupied_wires` (as per `Includes everything between used wires in a multi-wire gate.`)?

Reply:
yep.  At least depending on wire order.  Operation wires are first mapped to successive positive integers.
Because the drawn CNOT gate would be on top of 0, 1, and 2, all three wires are considered "occupied".